pmd-java/src/main/resources/category/java/bestpractices.xml

```diff
@@ -802,6 +802,76 @@ public class MyTest {
         </example>
     </rule>
 
+    <rule name="JUnit5TestShouldBePackagePrivate"
+          language="java"
+          since="6.35"
+          message="JUnit 5 tests that use @Test, @RepeatedTest, @TestFactory, @TestTemplate or @ParameterizedTest should be package private."
+          class="net.sourceforge.pmd.lang.rule.XPathRule"
+          typeResolution="true"
+          externalInfoUrl="${pmd.website.baseurl}/pmd_rules_java_bestpractices.html#junit5testshouldbepackageprivate">
+        <description>
+            JUnit 5 tests should be package private at the class level and for each method that uses @Test, @RepeatedTest, @TestFactory, @TestTemplate or @ParameterizedTest.
+        </description>
+        <priority>3</priority>
+        <properties>
+            <property name="version" value="2.0"/>
+            <property name="xpath">
+                <value>
+                    <![CDATA[
+//ClassOrInterfaceDeclaration[
+       matches(@SimpleName, $testClassPattern) and
+       @Abstract=false() and
+       @PackagePrivate=false()]
+       [ClassOrInterfaceBody/ClassOrInterfaceBodyDeclaration
+        [MethodDeclaration]
+        [Annotation//Name[
+            pmd-java:typeIs('org.junit.jupiter.api.Test') or pmd-java:typeIs('org.junit.jupiter.api.RepeatedTest')
+            or pmd-java:typeIs('org.junit.jupiter.api.TestFactory') or pmd-java:typeIs('org.junit.jupiter.api.TestTemplate')
+            or pmd-java:typeIs('org.junit.jupiter.params.ParameterizedTest')
+        ]]]
+|
+//ClassOrInterfaceDeclaration[
+       matches(@SimpleName, $testClassPattern)]
+    /ClassOrInterfaceBody/ClassOrInterfaceBodyDeclaration
+        [MethodDeclaration[@PackagePrivate=false()]]
+        [Annotation//Name[
+            pmd-java:typeIs('org.junit.jupiter.api.Test') or pmd-java:typeIs('org.junit.jupiter.api.RepeatedTest')
+            or pmd-java:typeIs('org.junit.jupiter.api.TestFactory') or pmd-java:typeIs('org.junit.jupiter.api.TestTemplate')
+            or pmd-java:typeIs('org.junit.jupiter.params.ParameterizedTest')
+        ]]
+]]>
+                </value>
+            </property>
+            <property name="testClassPattern" type="Regex" description="The regex pattern used to identify test classes" value="Test" />
+            <property name="version" value="2.0"/>
+        </properties>
+        <example>
+            <![CDATA[
+class MyTest {
```

Comment:
Please add comments in the example so that we know what to look at. Eg
```suggestion
class MyTest { // not public, that's fine
```

You could also reduce the size of the example by removing the body of methods and making them single-line. Eg
```java
    @Test
    public void testBad() {  } // should not have a public modifier
```
and so on

Reply:
Done